Users/abehenderson/CSX/Closures/hobbyTracker.js

```diff
@@ -0,0 +1,86 @@
+
+function hobbyTracker(hobbies) {
+  let cache = {};
+  for (let i = 0; i < hobbies.length; i++) {
+    cache[hobbies[i]] = 0;
+  }
+  
+  return function newFunc(hobby, hours) {
+    if (hobby === undefined && hours === undefined) {
+      for (const keys in cache) {
+        cache[keys] = 0;
+      }
+      return 'tracker has been reset!'
+    } else {
+      for (const property in cache) {
+         if (hobby === property) cache[property] += hours;
+      }
```

Comment:
```suggestion
     cache[property] += hours; // you don't need to iterate through the cache, keys are instantly accessible
     return cache
      }
```

Reply:
I'm not sure I understand how that would work.  `property` is declared in the `for...in` loop.  How else would I examine all of the keys to see if they match `hobby`?

Reply:
That's the point of a JS object: you don't have to. I've edited my suggestion above to remove the comment so it's clearer. To access a property on an object, just call it directly: `cache[hobby] += hours`.  No need to loop.